src/main/java/nl/ordina/jobcrawler/controller/LocationController.java

```diff
@@ -32,19 +33,20 @@ public LocationController(LocationService locationService, LocationModelAssemble
         this.locationModelAssembler = locationModelAssembler;
     }
 
-    @GetMapping
-    public Object getLocationByCoordinates(@RequestParam(required = false) Optional<Double> lat,
-                                           @RequestParam(required = false) Optional<Double> lon) throws IOException {
-        if(lat.isEmpty()&&lon.isEmpty()) {
-            return getLocations();
-        } else {
-            String location = locationService.getLocation(lat.get(), lon.get());
-            return ResponseEntity.status(HttpStatus.OK).body(Map.of(
-                    "success", true,
-                    "location", location));
+    @GetMapping("/coordinates")
+    public ResponseEntity<Map<String, Serializable>> getLocationByCoordinates(
+            @RequestParam Optional<Double> lat,
+            @RequestParam Optional<Double> lon) throws IOException {
+        String location = "";
+        boolean result = false;
+        if (lat.isPresent() && lon.isPresent()) {
+            location = locationService.getLocation(lat.get(), lon.get());
+            result = true;
         }
+        return ResponseEntity.status(HttpStatus.OK).body(Map.of("success", result, "location", location));
```

Comment:
Just wondering, do we need to return `HttpStatus.OK` if the success is set to false? If there is no location for the given coördinates it might be better to return `HttpStatus.NOT_FOUND`.
https://stackoverflow.com/questions/11746894/what-is-the-proper-rest-response-code-for-a-valid-request-but-an-empty-data

Reply:
Heb 'm veranderd naar NOT_FOUND als ie niet gevonden kan worden